tests/integration/security/jwt_test.go

```diff
@@ -62,335 +61,284 @@ func TestRequestAuthentication(t *testing.T) {
-									Target:   dest[0],
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Count:    callCount,
-								},
-								DestClusters: dest.Clusters(),
+						ExpectResponseCode: response.StatusUnauthorized,
+					},
+					{
+						Name:   "invalid-jwks-expired-token-noauthz",
+						Config: "invalid-jwks",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Headers: map[string][]string{
+								authHeaderKey: {"Bearer " + jwt.TokenExpired},
 							},
-							ExpectResponseCode: response.StatusCodeOK,
+							Count: callCount,
 						},
-					}
-					for _, c := range testCases {
-						t.NewSubTest(c.Name).Run(func(t framework.TestContext) {
+						ExpectResponseCode: response.StatusUnauthorized,
+					},
+					{
+						Name:   "invalid-jwks-no-token-noauthz",
+						Config: "invalid-jwks",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Count:    callCount,
+						},
+						ExpectResponseCode: response.StatusCodeOK,
+					},
+				}
+				for _, c := range testCases {
+					echotest.New(t, apps.All).
+						SetupForPair(func(t framework.TestContext, src, dst echo.Instances) error {
 							if c.Config != "" {
-								policy := tmpl.EvaluateOrFail(t,
-									file.AsStringOrFail(t, fmt.Sprintf("testdata/requestauthn/%s.yaml.tmpl", c.Config)), namespaceTmpl)
-								t.Config().ApplyYAMLOrFail(t, ns.Name(), policy)
-								util.WaitForConfig(t, policy, ns)
+								policy := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
+									file.AsStringOrFail(t, fmt.Sprintf("testdata/requestauthn/%s.yaml.tmpl", c.Config)),
+									map[string]string{
+										"Namespace": ns.Name(),
+										"dst":       dst[0].Config().Service,
+									},
+								), ns.Name())
+								return t.Config().ApplyYAML(ns.Name(), policy)
 							}
-
-							retry.UntilSuccessOrFail(t, c.CheckAuthn, echo.DefaultCallRetryOptions()...)
+							return nil
+						}).
+						From(
+							// TODO(JimmyCYJ): enable VM and fix valid-token-forward-remote-jwks and invalid_aud.
+							filters(t, ns.Name(), true)...).
+						ConditionallyTo(echotest.ReachableDestinations).
+						ConditionallyTo(func(from echo.Instance, to echo.Instances) echo.Instances {
+							return to.Match(echo.Not(echo.SameDeployment(from)))
+						}).
+						To(filters(t, ns.Name(), false)...).
+						Run(func(t framework.TestContext, src echo.Instance, dest echo.Instances) {
+							t.NewSubTest(c.Name).Run(func(t framework.TestContext) {
+								c.CallOpts.Target = dest[0]
+								c.DestClusters = dest.Match(echo.InCluster(src.Config().Cluster)).Clusters()
+								c.CallOpts.Validator = echo.And(echo.ValidatorFunc(c.CheckAuthn))
+								src.CallWithRetryOrFail(t, c.CallOpts, echo.DefaultCallRetryOptions()...)
+							})
 						})
-					}
-				})
-			}
+				}
+			})
 		})
 }
 
+func filters(t framework.TestContext, ns string, skipVM bool) []echotest.Filter {
```

Comment:
Done. Thanks! 
The pattern is to always explicitly exclude unwanted workloads, so that we don't need to add new types of workload in the future. I add a TODO to simply this method. Let me know if you have any idea to use single filter.